### ISSUE TITLE
Upgrade to clFFT and clBLAS

### ISSUE THREAD

**Issue by GoogleCodeExporter** [adda-team/adda#204]
```
AMD has open-sourced its projects clAmdFft and clAmdBlas some time ago. Now 
they are called clFFT and clBLAS respectively. And these projects have new 
binary releases:
https://github.com/clMathLibraries/clFFT/releases
https://github.com/clMathLibraries/clBLAS/releases

The changelog is probably not that impressive yet, but it make sense to upgrade 
to it before the next release. Then the user will be able to benefit from 
further development of clFFT and clBLAS simply by downloading new DLLs from 
this projects.

The upgrade should be a matter of removing "Amd" from all names of headers, 
libraries, and functions. But also wiki-s with installation instructions for 
these packages should be updated.

```

Original issue reported on code.google.com by `yurkin` on 3 Aug 2014 at 4:49
- Blocking: #157

**Comment by myurkin** [comment]
So, #211 fixes the renaming part. The rest is:
1) updating wikis, but that we should do after the wiki migration is complete (#210). Related issue is the linking paths on Windows.

2) Understanding whether the current version of clFFT and clBLAS has any new features that we can use. The possible ideas are mentioned in the comments inside the source files, including:
* use of `CLFFT_DOUBLE_FAST`
* advanced FFT plans (like `guru...` in FFTW)
* dependency on `pthread` on Unix
* including version headers (whether the main header is enough)

Addressing (2) should probably require reading the current manuals for clFFT - http://clmathlibraries.github.io/clFFT/ (I can't find anything comparable for clBLAS). @mapclyps , can you please look at it? I've seen your comment about `pthread`, but we will need some tests on other systems as well (and it's safe to leave this option just-in-case anyway)

**Comment by mapclyps** [comment]
Nice that you collected all the comments regarding clFFT and clBLAS from our source files!

There is a documentation for all function of clBLAS at http://clmathlibraries.github.io/clBLAS/. I used this documentation to infer what clBLAS functions were actually needed to do the same operations as executed in the BiCG solver on the vectors.

I just looked shortly into all points except the version headers

* `CLFFT_DOUBLE_FAST` is still disabled in the current version of clFFT.

* I'm not familiar with the `guru...`-plans in FFTW but from what I got from your comments in `fft.c` it may be achieved with strides as described in http://clmathlibraries.github.io/clFFT/index.html#DistanceStridesandPitches 
I can have a look at this some when. However, the performance impact seem not too strong currently so I suggest to add this as a new issue with low priority.

* Regarding the `pthread` issue, I think it is resolved now in the clFFT library as they added linking of pthread now to their Makefile.
https://github.com/clMathLibraries/clFFT/pull/37
So I think `-lpthread` option should not be needed anymore when compiling adda with the current clFFT version even though `pthread` is still needed. However, I don't know how this works on Windows platforms and such.

* The current supported radix sizes by clFFT are 2,3,5 with the plan to add 7. 
https://github.com/clMathLibraries/clFFT/issues/1

**Comment by myurkin** [comment]
I will test pthread on Windows (just in case) and then remove this line from Makefile (somewhen) .
@mapclyps, can you please look if #157 is still relevant?

**Comment by myurkin** [comment]
Now we have decent wiki pages - https://github.com/adda-team/adda/wiki/InstallingclFFT and https://github.com/adda-team/adda/wiki/InstallingclBLAS

**Comment by myurkin** [comment]
Fixed with commit afcfc94.

The remaining issue is #214 and `CLFFT_DOUBLE_FAST`. However, the latter may never appear since clFFT seems to be dormant (not further developed).
